### DISCUSSION TITLE
Blazor Binding a View Parameter Property to a View Model Property

### DISCUSSION THREAD

**Discussion by TorreyGarland** [reactiveui/ReactiveUI#3325] [Q&A]
What is the easiest way to bind a query string parameter property to a view model property.

this.whenActivated and this.bind/onewaybind do not work.

**Comment by jaredthirsk** [comment]
Have you tried changing the Parameter's setter to call OnPropertyChanged()?

**Reply by TorreyGarland** [reply]
```
Have you tried changing the Parameter's setter to call OnPropertyChanged()?
```

Current version of Visual Studio displays a warning:

![warning-2](https://github.com/reactiveui/ReactiveUI/assets/1054548/b6bad711-d32a-404d-a667-f763807e4375)

using Visual Studio Community 2022 17.8.3... non-preview version.

What now???

**Reply by jaredthirsk** [reply]
Well, that was the easiest.  The warning advises that that sort of thing could cause problems -- I'm not sure if it would in this case.  If not, you could disable the warning.

Here is a less easy approach that abides by the rule of having auto properties for [Parameter]s:

https://stackoverflow.com/a/64419271/208304

Edit:  I believe the idea here is that some parent of the Razor component owns the value, and that's where ReactiveUI should be bound.  I'm not sure this is most desirable for your case though.

@TorreyGarland
